### ISSUE TITLE
Upgrade package

### ISSUE THREAD

**Issue by ratik21** [MarcoPolo/y-libp2p#1]
Hi @MarcoPolo , libp2p has released various updates and changes (some breaking). Will this repo be updated to be used with the latest libp2p api's? We're using this in our backend.

**Comment by MarcoPolo** [comment]
Yeah it would be great, especially to incorporate the esm changes. 

Want to give this a shot?

**Comment by ratik21** [comment]
Yes, I am trying to do it, needed some help. I have upgraded but still having issue with the real time sync b/w two nodes. Let me create a PR, thanks :)
